### PR TITLE
 hamming: Updated the exercise to the 2.1.1 version

### PR DESCRIPTION
**Pull request by ZapAnton**
Relevant PRs:
    - https://github.com/exercism/problem-specifications/issues/625
    - https://github.com/exercism/problem-specifications/pull/844
    - https://github.com/exercism/problem-specifications/pull/845
    - https://github.com/exercism/problem-specifications/pull/875
    - https://github.com/exercism/problem-specifications/pull/953
    - https://github.com/exercism/problem-specifications/pull/1129
    - https://github.com/exercism/problem-specifications/pull/1389

Basically adds every test case from the current canonical-data.json  without deleting the old ones.

### CODE REVIEW

File: exercises/hamming/tests/hamming.rs [hunk 1]
```diff
@@ -1,14 +1,21 @@
 extern crate hamming;
 
+fn process_distance_case(strand_pair: [&str; 2], expected_distance: Option<usize>) {
```
Comment (coriolinus) [inline exercises/hamming/tests/hamming.rs]: I'm curious why you wrote this as `[&str; 2]` instead of `(&str, &str)`. I tend to think of function arguments as tuple-like instead of array-like, so the latter would have seemed more natural to me.

I suspect that the two forms are equivalent in memory, and this does work; you certainly don't need to change it. I'm just curious about your design process.

Comment (ZapAnton) [reply exercises/hamming/tests/hamming.rs]: Guess I decided to go with the array, because the types of the variables are similar (perhaps a little C-like design).

If the strands where represented with different types, e.g. `LeftStrand` / `RightStrand`, then I would surely used the tuple.

**Comment by petertseng** [comment]
Since this also updates the README, https://github.com/exercism/problem-specifications/pull/1360 is now also relevant.
